sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1753,6 +1753,27 @@ class Dataset[T] private[sql](
    *
    * Also as standard in SQL, this function resolves columns by position (not by name).
    *
+   * Notice that the column positions in the schema aren't necessarily matched with the
+   * fields in the typed objects in a Dataset. This function resolves columns by their positions
+   * in the schema, not the fields in the typed objects, as this Scala example shows:
+   *
+   * {{{
+   *   case class Test(a: String, b: String)
+   *   val ds1 = Seq(("a", "b")).toDF("a", "b").as[Test] // ds1's schema: [a: String, b: String]
+   *   val ds2 = Seq(("b", "a")).toDF("b", "a").as[Test] // ds2's schema: [b: String, a: String]
+   *   ds1.union(ds2).show
+   *
+   *   // output:
+   *   // +---+---+
+   *   // |  a|  b|
+   *   // +---+---+
+   *   // |  a|  b|
+   *   // |  b|  a|
+   *   // +---+---+
```

Comment:
Please use the same example as `union `. Just need to add a comment to explain it is also applicable to the strongly-typed JVM objects.

Reply:
Sorry, I don't get the meaning of same example as `union`. This is the only example of `union`, if I don't miss anything.

Reply:
Added a comment to say it's also applicable to the strongly-typed JVM objects.

Reply:
Please use the same example as `union`. Users can easily get the difference between `union` and `unionByName`.

Reply:
Sorry, but this is the example of `union`. Which example you mean to use here? Are you meaning to use the example of `unionByName`?

Reply:
If I understand your comment correctly, you mean we don't need to add this example for explaining how `union` works on typed objects. Just a normal union example is good.

I'm ok for it, although from past experiences, there are many end users getting confused by this difference.